test/test_nn.py

```diff
@@ -13261,6 +13263,134 @@ def test_conv3d_valid_padding_backward(self, device):
+                decorators=[onlyCPU, disableMkldnn], name='empty_batch2d'),
+        subtest(((2, 0, 7, 8), False, False, 3, torch.strided, torch._C.ConvBackend.Empty),
+                decorators=[onlyCPU, disableMkldnn], name='empty_channel2d'),
+        subtest(((0, 0, 7, 8), False, False, 3, torch.strided, torch._C.ConvBackend.Empty),
+                decorators=[onlyCPU, disableMkldnn], name='empty_batch_channel2d'),
+        subtest(((0, 6, 7, 8, 9), False, False, 3, torch.strided, torch._C.ConvBackend.Empty),
+                decorators=[onlyCPU, disableMkldnn], name='empty_batch3d'),
+        subtest(((2, 0, 7, 8, 9), False, False, 3, torch.strided, torch._C.ConvBackend.Empty),
+                decorators=[onlyCPU, disableMkldnn], name='empty_channel3d'),
+        subtest(((0, 0, 7, 8, 9), False, False, 3, torch.strided, torch._C.ConvBackend.Empty),
+                decorators=[onlyCPU, disableMkldnn], name='empty_batch_channel3d'),
+        # === cuda ===
+        # Note that disablecuDNN disables miopen as well.
+        subtest(((2, 6, 7), False, False, 6, torch.strided, torch._C.ConvBackend.CudaDepthwise2d),
+                decorators=[onlyCUDA, disablecuDNN], name='cuda_depthwise1d'),
+        subtest(((2, 6, 7, 8), False, False, 6, torch.strided, torch._C.ConvBackend.CudaDepthwise2d),
+                decorators=[onlyCUDA, disablecuDNN], name='cuda_depthwise2d'),
+        subtest(((2, 6, 7, 8, 9), False, False, 6, torch.strided, torch._C.ConvBackend.CudaDepthwise3d),
+                decorators=[onlyCUDA, disablecuDNN], name='cuda_depthwise3d'),
+        # === cudnn ===
+        subtest(((2, 6, 7), False, False, 3, torch.strided, torch._C.ConvBackend.Cudnn),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn1d'),
+        subtest(((2, 6, 7, 8), False, False, 3, torch.strided, torch._C.ConvBackend.Cudnn),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn2d'),
+        subtest(((2, 6, 7, 8, 9), False, False, 3, torch.strided, torch._C.ConvBackend.Cudnn),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn3d'),
+        subtest(((2, 6, 7), True, False, 3, torch.strided, torch._C.ConvBackend.CudnnTranspose),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn1d_transposed'),
+        subtest(((2, 6, 7, 8), True, False, 3, torch.strided, torch._C.ConvBackend.CudnnTranspose),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn2d_transposed'),
+        subtest(((2, 6, 7, 8, 9), True, False, 3, torch.strided, torch._C.ConvBackend.CudnnTranspose),
+                decorators=[onlyCUDA, skipCUDAIfNoCudnn], name='cudnn3d_transposed'),
+        # === mkldnn ===
+        # 1D conv is not supported for mkldnn tensors
+        subtest(((2, 6, 7), False, False, 3, torch._mkldnn, torch._C.ConvBackend.Mkldnn),
+                decorators=[onlyCPU, skipCPUIfNoMkldnn, expectedFailure('cpu')], name='mkldnn1d'),
```

Comment:
expected failure?

Reply:
Found that 1D mkldnn inputs fail with the message: "opaque tensors do not have strides".

An explicit check for this case should be added - it'd be nice to have `assertRaisesRegex` in decorator form for testing this

Reply:
Ok!
Maybe link the corresponding issue here in case a future reader is wondering.

Reply:
Linked the issue! #68034